source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -187,6 +204,10 @@ def _getWordOffsetsInThisLine(self, offset, lineInfo):
 			min(end.value, max(1, len(lineText) - 2))
 		)
 
+	def __ne__(self,other):
+		"""Support more accurate caret move detection."""
```

Comment:
How does __ne__ relate to caret move detection?

Reply:
Can you explain exactly what happens if you don't implement this method?

Reply:
```__ne__``` is necessary as UIATextInfo has overridden ``__eq__```. So we want ```__ne__``` to return the opposite. This should really actually be on UIATextInfo itself, but this has caused strange bugs in the past, espcially in Edge. We are not sure why IUIAutomationTextRange.compare does not work correctly for some implementations.  In consoles however, it does work okay.
Note that in Python3 it will no longer be neceesary to provide a ```__ne__``` here as Python3  automatically implements ```__ne__`` as the negative of ```__eq__```.